File: Stack/PHP/Stack.php
<?php declare(strict_types = 1);

class Stack {

    protected array $stack;
    protected int $capacity;
    protected int $size;
    protected string $itemType;

    public function __construct(int $capacity, string $itemType = '') {

        if ($capacity <= 0) {
            throw new InvalidArgumentException('Capacity should be greater than 0.');
        }

        $this->stack = [];
        $this->size = 0;
        $this->capacity = $capacity;
        $this->itemType = $itemType;

    }

    public function size(): int {
        return $this->size;
    }

    public function isEmpty(): bool {
        return $this->size <= 0;
    }

    public function isFull(): bool {
        return $this->size >= $this->capacity;
    }

    public function isValidType($item): bool {
        return gettype($item) === $this->itemType;
    }

    public function print() {
        var_dump($this->stack);
    }

    public function push($item): void {

        if (!$this->isValidType($item)) {
            throw new InvalidArgumentException('Expecting ' . $this->itemType . ', but got ' . gettype($item) . '.');
        }

        if ($this->isFull()) {
            throw new RuntimeException('The Stack is overflow. It is impossible to add new element');
        }

        $this->stack[] = $item;
        $this->size++;

    }

    public function pop()  {

        if ($this->isEmpty()) {
            throw new RuntimeException('The Stack is empty. It is impossible to remove element.');
        }

        $item = $this->stack[$this->size - 1];
        unset($this->stack[$this->size--]);
        return $item;

    }

    public function peek() {

        if ($this->isEmpty()) {
            return null;
        }

        return $this->stack[$this->size - 1];

    }

}

File: Stack/Java/Stack.java
import java.util.NoSuchElementException;

public class Stack {

    private int stack[];
    private int capacity;
    private int top;

    Stack(int capacity) {

        if(capacity <= 0) {
            throw new IllegalArgumentException("Capacity should be greater than 0.");
        }

        this.capacity = capacity;
        this.stack = new int[capacity];
        this.top = 0;

    }

    public int size() {

        return this.top;

    }

    public boolean isEmpty() {

        return this.size() == 0;

    }

    public boolean isFull() {

        return this.size() >= this.capacity;
        
    }

    public void push(int item) {

        if (this.isFull()) {
            throw new StackOverflowError("The Stack is overflow. It is impossible to add new element");
        }

        this.stack[this.top++] = item;

    }

    public int pop() {

        if (this.isEmpty()) {
            throw new NoSuchElementException("The Stack is empty. It is impossible to remove element.");
        }

        return this.stack[--this.top];

    }

    public int peek() {

        if (this.isEmpty()) {
            throw new NoSuchElementException("The Stack is empty. It is impossible to peek element.");
        }

        return this.stack[this.top - 1];

    }

    public void print() {

        for(int i = 0; i < this.top; i++) {
            System.out.println(i + " - " + this.stack[i]);
        }

    }

}